utils/field.js

```diff
@@ -98,6 +99,19 @@ const generateFakeDataForField = (field, faker, changelogDate, type = 'csv') =>
   if (field.fakerTemplate) {
     data = faker.faker(field.fakerTemplate);
   } else if (field.fieldValidate && field.fieldValidateRules.includes('pattern')) {
+    // check if regex is valid. If not, issue warning and we skip fake data generation.
+    try {
+      // eslint-disable-next-line no-new
+      new RegExp(field.fieldValidateRulesPattern);
+    } catch (e) {
+      console.log(
+        `${chalk.yellowBright('WARNING!')} ${field.fieldName} pattern is not valid: ${
+          field.fieldValidateRulesPattern
+        }. Skipping generating fake data. `
+      );
+      return undefined;
+    }
+
     const generated = field.createRandexp().gen();
```

Comment:
```suggestion
    const dataGenerator = field.createRandexp();
    if (!dataGenerator) {
      return undefined;
    }
    const generated = dataGenerator.gen();
```

- Move the regex check to:
https://github.com/jhipster/generator-jhipster/blob/cbd88e4daa6681c2a512c8a73c6621e156564638/utils/field.js#L374
like
```
    field.createRandexp = () => {
      // check field.fieldValidateRulesPattern and return undefined.
      return faker.createRandexp(field.fieldValidateRulesPattern);
    }
```

- Replace console.log with this.warning.

Reply:
Done